### PR TITLE
Surprise!

### PR DESCRIPTION
**Pull request by eugeneyche**
No one knows what this does.

### CODE REVIEW

File: src/Fx.tsx [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    let lastRenderMs = Date.now();
+    const loopImpl = () => {
+        if (!fx.active) {
+            return;
+        }
+        const nowMs = Date.now();
+        const timeDeltaMs = nowMs - lastRenderMs;
+        lastRenderMs = nowMs;
+        renderFxFrame(fx, timeDeltaMs / 1_000);
+        setTimeout(loopImpl, 1_000 / FX_FPS)
```
Comment (goffrie) [inline src/Fx.tsx:219]: This should be using requestAnimationFrame

Comment (eugeneyche) [reply src/Fx.tsx:219]: hmmm, does that limit framerate to 60, or based on display refresh rate? I was hoping to always just limit to 60

Comment (goffrie) [reply src/Fx.tsx:219]: Why fps cap

Comment (eugeneyche) [reply src/Fx.tsx:219]: it's a silly effect; don't want it comsuming too much resources

Comment (goffrie) [reply src/Fx.tsx:219]: requestAnimationFrame also handles other edge cases like weird (or lower) refresh rates, doesn't run while offscreen, and provides a more accurate timestamp for smooth animation though.
You could maybe skip the frame if it's too close to the last one

File: src/Game.tsx [hunk 2]
```diff
@@ -363,6 +368,17 @@ function ScoringGame(props: ScoringGameProps) {
     const gameWon = useMemo(() => {
         return playerScores.every((p, i) => i === 0 || !pokerHandLessThan(p.score, playerScores[i - 1].score));
     }, [playerScores]);
+    const { shootConfetti } = useContext(FxContext);
```
Comment (goffrie) [inline src/Game.tsx:371]: Why does this need a context and not just a hook?

File: src/Game.tsx [hunk 3]
```diff
@@ -363,6 +368,17 @@ function ScoringGame(props: ScoringGameProps) {
...
+    useEffect(() => {
+        // Intentionally do this when the game is unwonnered.
+        // That way, no one will find out about this, since we always win.
+        if (revealIndex === players.length && !gameWon) {
+            shootConfetti(0, window.innerHeight, -Math.PI / 2 + 0.4);
+            const timeoutHandle = setTimeout(() => {
```
Comment (goffrie) [inline src/Game.tsx:377]: unused variable?

File: src/Game.tsx (hunk 2 above)
Comment (goffrie) [reply src/Game.tsx:371]: Or rather, it should probably just be a component that is rendered only if the game is not wonnered

File: src/Fx.tsx (hunk 1 above)
Comment (eugeneyche) [reply src/Fx.tsx:219]: ah good idea

**Pull request merged**